packages/dds/tree/src/tree/anchorSet.ts

```diff
@@ -104,41 +104,124 @@ export class AnchorSet {
 
     /**
      * Updates paths for a range move (including re-parenting path items and updating indexes).
+     * @param count - number of siblings to to insert/delete/move.
+     * @param src - where the siblings are removed from. If undefined the operation is an insert.
+     * @param dst - where the siblings are moved to. If undefined the operation is a delete.
+     *
+     * TODO:
+     * How should anchors that become invalid, then valid again (ex: into content that was deleted, then undone) work?
+     * Add an API to resurrect them? Store them in special detached fields? Store them in special non-detached fields?
```

Comment:
Good question. @alex-pardes, this is making me think Anchors should perhaps have some gap lineage information on them. What do you think?

Reply:
I think there at least two related issues here:
- Place style anchors, which don't always move with nodes. I think we can express these as node anchors to the parent + some extra anchor type specific info.
- Node anchors when the nodes they are tracking go away, then come back. To make this efficient for large subtrees (not have to touch each anchor in the deleted subtree), I think we want to leave the subtree's PathNodes intact, and the current way to do this is a detached field.

My thought is that fancy cell/gap lineage logic type stuff falls into the first category and can be done via custom anchor types which the rebaser has special knowledge of (since it creates them) and can thus update/maintain accordingly.

It would also be possible for the Rebaser to control the AnchorSet implementation (turning the type in tree into an interface instead of a concretion), but for now I think we can avoid that.